ha/fault_tolerance/rest_api.py

```diff
@@ -0,0 +1,255 @@
+import signal
+import ssl
+import time
+from abc import ABC
+from aiohttp import web
+import asyncio
+from asyncio import CancelledError as AsyncioCancelledError
+from concurrent.futures import CancelledError as ConcurrentCancelledError
+
+from cortx.utils.log import Log
+from cortx.utils.conf_store import Conf
+from ha import const
+
+# TODO : CORTX-30931 to be replace with full definition
+class Response:
+    pass
+
+class CcRestApi(ABC):
+
+    _app = None
+    _loop = None
+    _site = None
+    __is_shutting_down = False
+    _handle_signals = True
+    _signals = ('SIGINT', 'SIGTERM')
+
+    # Test function just to check whether CC REST API server is alive
+    # TODO: to be removed CORTX-30932
+    @staticmethod
+    async def handler_check_alive(request):
+        """
+        Test Handler function is added for testing purpose.
+        it returns test "CORTX CC REST API server is alive."
+        Args:
+            request (web.Request): Request object
+        Returns:
+            web.Response: response object that to be sent to client
+        """
+        return web.Response(text="CORTX CC REST API server is alive.")
+
+    @staticmethod
+    def init(handle_signals=True) -> None:
+        """
+        Initialize web application ans set supported routes, middlewares, etc.
+        Args:
+            handle_signals (bool, optional): Defaults to True.
+            If handle_signals is True then signal handler will be set.
+            if it is false then caller should handle the signals.
+            and needs to call CcRestApi.stop() before application terminates
+        """
+
+        CcRestApi._handle_signals = handle_signals
+
+        # Create Rest Application object and set middleware coroutine to it.
+        # A middleware is a coroutine that can modify either the request or response.
+        CcRestApi._app = web.Application(middlewares=[CcRestApi.rest_middleware])
+
+        # TODO: CORTX-30932 Add routs using view classes and remove below placeholder
+        # Note: Adding route '/' just to test
+        CcRestApi._app.router.add_get('/', CcRestApi.handler_check_alive)
+
+        CcRestApi._app.on_startup.append(CcRestApi._on_startup)
+        CcRestApi._app.on_shutdown.append(CcRestApi._on_shutdown)
```

Comment:
What is the purpose of these two?

Reply:
on_startup  : Subscribers may use the signal to run background tasks in the event loop along with the application’s request handler just after the application start-up.
For more info please refer : https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.Application.on_startup